addons/osfstorage/decorators.py

```diff
@@ -96,7 +96,12 @@ def wrapped(payload, *args, **kwargs):
             user = OSFUser.load(payload['user'])
             # Waterbutler is sending back ['node'] under the destination payload - WB should change to target
             target = payload['destination'].get('target') or payload['destination'].get('node')
-            dest_target = Guid.load(target).referent
+            if VersionedGuidMixin.GUID_VERSION_DELIMITER in target:
+                dest_target = GuidVersionsThrough.objects.filter(
+                    guid___id=target.split(VersionedGuidMixin.GUID_VERSION_DELIMITER)[0]
+                ).first().referent
+            else:
+                dest_target = Guid.load(target).referent
```

Comment:
If we have a versioned guid, we need to find the referent for the exact version.

```python
# TODO: turn this into a helper method `get_referent_from_versioned_guid()`
guid_split_list = target.lower().split(VersionedGuidMixin.GUID_VERSION_DELIMITER)
guid = guid_split_list[0]
version = guid_split_list[1] if len(guid_split_list) > 1 else None
if version:
    dest_target = GuidVersionsThrough.objects.get(guid___id=guid, version= version).referent
else:
    dest_target = Guid.load(guid).referent
```

Note: must use `target.lower()` before splitting the `versioned_guid`
Note: let's use `get()` instead of `filter().first()` since we will add uniqueness constraints later for `(guid, version)`.

Reply:
Changed it